Added a buildpack registry extension (#79)
buildpacks/spec: extensions/buildpack-registry.md

## Backing Storage

The registry information is stored in a repository called an index. The index is a set of files and directories that contain information about available buildpacks.

The registry index MUST be stored in a Git repository.

The index MUST work across all major Operating Systems.

### Directory Structure

Folders in the index MUST be split by two nested folders. The buildpack ID will determine the name of the folders where:
* The first folder MUST represent the first two characters
* The second folder MUST represent the third and fourth characters
* IDs that are 1-3 characters long MUST be represented by special folders.

The folders will contain files that represent each buildpack ID where the following is true:
* The `/` character in the ID MUST be replaced with `_`
* The namespace MUST match `[a-z0-9\-\.]{1,253}`
* The name MUST match `[a-z0-9\-\.]{1,253}`

Here's an example directory structure:

```
1
├── example_a
└── example_b
2
├── example_aa
└── example_ab
3
├── a
│   ├── foobar_abc
│   └── example_acd
└── b
    ├── example_bcd
    └── example_bed
fo
├── ob
│   ├── samples_fooball
│   └── example_foobar
└── oc
    └── example_foocal
```

> The following ids are reserved by Windows, so they aren't allowed as valid ids:

Reviewer: Probably should have asked about this on the RFC, but I don't understand this part.

Reviewer 2: @ekcasey I'm trying to remember my discussion with Sean from the crates.io team, but these names are reserved by windows as filenames.

Reviewer 2: From the last leadership meeting: Due to the `_` after 2 characters, these names will never come up. Let's cut this section.

Reviewer 2: @jkutner do you want to make this change?

Author: done

* nul
* con
* prn
* aux
* com1
* com2
* com3
* com4
* com5
* com6
* com7
* com8
* com9
* lpt1
* lpt2
* lpt3
* lpt4
* lpt5
* lpt6
* lpt7
* lpt8
* lpt9

### File Contents

> Each file in the index MUST represent a buildpack. The file MUST contain multiple entries representing each version of the buildpack split by a newline. Each version entry:

Reviewer: ```suggestion
Each file in the index MUST represent a buildpack. The file MUST contain one or more entries representing each version of the buildpack, delimited by a newline. Each version entry:
```
